HannaPidchasiuk/HW10/worker.py

```diff
@@ -0,0 +1,169 @@
+        """
+        self.__last_name = new_value
+
+    @property
+    def salary(self):
+        """
+            Returns salary property of the worker.
+        """
+        return self.__salary
+
+    @salary.setter
+    def salary(self, new_value: int):
+        """
+            Set new salary property of the worker.
+            Takes only 1 argument: new_value.
+        """
+        if new_value < 0:
+            raise ValueError("Salary can't be less than zero.")
+        else:
+            self.__salary = new_value
+
+    @property
+    def department(self):
+        """
+            Returns department property of the worker.
+        """
+        return self.__department
+
+    @department.setter
+    def department(self, new_value: Department):
+        """
+            Set new department property of the worker.
+            Takes only 1 argument: new_value.
+        """
+        self.__department = new_value
+
+    @property
+    def grade(self):
+        """
+            Returns grade property of the worker.
+        """
+        return self.__grade
+
+    @grade.setter
+    def grade(self, new_value: Grade):
+        """
+            Set new grade property of the worker.
+            Takes only 1 argument: new_value.
+        """
+        self.__grade = new_value
+
+    @property
+    def type_of_work(self):
+        """
+            Returns type of work for worker.
+        """
+        return self.__type_of_work
+
+    @type_of_work.setter
+    def type_of_work(self, new_value: TypeOfWork):
+        """
+            Set new type of work for the worker.
+            Takes only 1 argument: new_value.
+        """
+        self.__type_of_work = new_value
+
+    @property
+    def date_of_start(self):
+        """
+            Returns first working day date.
+        """
+        return self.__date_of_start
+
+    @date_of_start.setter
+    def date_of_start(self, new_value: datetime):
+        """
+            Set new first working day date.
+            Takes only 1 argument: new_value.
+        """
+        self.__date_of_start = new_value
+
+    def count_overtimes(self, overtime_hours: int, average_hours: float) -> float:
+        """
+            Function takes 2 arguments and count price for overtimes:
+                - overtime_hours - amount of hours to pay for;
+                - average_hours - average amount of working hours for current month
+        """
+        if not overtime_hours or not average_hours:
+            raise ZeroDivisionError("Working hours/overtime hours can't be 0!")
+        else:
+            return round((self.__salary / average_hours * overtime_hours), 2)
+
+    def count_working_period(self):
+        """
+            Counts period worker works in a company.
+            If less than 1 year only month number will be returned.
+            If less than 1 month - number of working days will be returned.
+        """
+        period = datetime.now() - self.__date_of_start
```

Comment:
what if self.__date_of_start will be in the future. 
It could happen actually. Please handle this case